### PR TITLE
Add Github action to run `unittest`

### PR DESCRIPTION
**Pull request by laserkelvin**
# Related issues

Partially addresses #47, adding `unittest` as part of checks.

# Proposed changes

- Add a Github action to run `unittest` after installing CBI
- Supports Python version matrix, which can be updated later to add more Python versions to test against

Currently because of #63 one of the tests (`test_cbiconfig_file`) will fail.

### CODE REVIEW

File: .github/workflows/run-unittest.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: CBI unittest
+
+on: [push]
```
Comment (Pennycook) [inline .github/workflows/run-unittest.yml:3]: The equivalent file in the [P3 Analysis Library](https://github.com/intel/p3-analysis-library/blob/main/.github/workflows/build-unittest.yml) CI also runs on pull requests.  Do we need both here too?

Comment (Pennycook) [inline .github/workflows/run-unittest.yml:3]: Should we also limit the paths here?  I think we probably only need to run the tests if something in `bin/*`, `codebasin/**`, `setup.py` or `MANIFEST.in` changes.  Without limiting the paths, won't we also run all the tests when we change something like the documentation?

File: .github/workflows/run-unittest.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (Pennycook) [inline .github/workflows/run-unittest.yml:4]:
```suggestion
permissions: read-all
```

File: .github/workflows/run-unittest.yml (hunk 1 above)
Comment (laserkelvin) [reply .github/workflows/run-unittest.yml]: Addressed in d1c985f

File: .github/workflows/run-unittest.yml (hunk 2 above)
Comment (laserkelvin) [reply .github/workflows/run-unittest.yml:24]: Addressed in 8e77d27

**Comment by Pennycook** [comment]
Let's see what happens... 🤞

**Pull request merged**